tests/network/link_state.go

```diff
@@ -0,0 +1,150 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"kubevirt.io/kubevirt/tests/libwait"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libnet"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = SIGDescribe("interface state up/down", func() {
+
+	Context("VM with one link up and  one link down", func() {
```

Comment:
I do not think this makes sense.
The context is about starting a VM with link-state set on startup.
I do not think the creation of the VM should be a fixture, it is part of the test.

Reply:
Let me refine my comment a bit.
I see no need for this "level" at all. The only setup I see here is the NAD creation which can be is the "Describe".
The VM fits more in the test (It) than in the setup. But you can try and explain me why you thing the VM creation is not the actual test itself.